### PR TITLE
v1.2.0

### PR DESCRIPTION
**Pull request by bear454**

### CODE REVIEW

File: app/models/concerns/authenticateable.rb [hunk 1]
```diff
@@ -13,17 +13,19 @@ def password_is_valid?(password)
     salt = salted_password.split("$")[2]
 
     salted_password == salt_password(salt, password)
-  rescue StandardError, Cheetah::ExecutionFailed
+  rescue StandardError, Cheetah::ExecutionFailed => e
```
Comment (jesusbv) [inline app/models/concerns/authenticateable.rb:16]: unused `e` ?

File: app/models/concerns/authenticateable.rb [hunk 2]
```diff
@@ -13,17 +13,19 @@ def password_is_valid?(password)
...
     return false
   end
 
   private
 
   def salt_password(salt, password)
-    Cheetah.run(
+    stdout, stderr = Cheetah.run(
```
Comment (jesusbv) [inline app/models/concerns/authenticateable.rb:23]: should we not check `stderr` ?

File: app/handlers/exceptions_app.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  def bad_request; end
+
+  def conflict; end
+
+  def internal_server_error
+    supportconfig = Supportconfig.new(dir: Rails.root.join('public'))
+    supportconfig.generate unless supportconfig.generated?
+    @supportconfig = '/' + supportconfig.output_filename
+  rescue StandardError => e
```
Comment (jesusbv) [inline app/handlers/exceptions_app.rb:12]:
```ruby
rescue StandardError, Cheetah::ExecutionFailed => e
```
missing the Cheetah exception (from `supportconfig.generate`)

File: app/models/supportconfig.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+  attr_reader :generated_at
+
+  def initialize(dir: Rails.root.join('public'))
+    @dir = dir
+    @generated_at = KeyValue.get('supportconfig_generated_at')
+  end
+
+  def generate()
+    stdout, stderr = Cheetah.run(
```
Comment (jesusbv) [inline app/models/supportconfig.rb:14]: ditto, check `stderr` ?

File: app/models/supportconfig.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    @generated_at = DateTime.now
+    KeyValue.set('supportconfig_generated_at', @generated_at)
+  end
+
+  def generated?()
+    !!@generated_at
+  end
+
+  def output_filename()
+    raise NotGeneratedError unless @generated_at
```
Comment (jesusbv) [inline app/models/supportconfig.rb:34]: consider adding an empty line after `unless`

File: app/models/supportconfig.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    'scc_' + self.filename_element + '.' + self.extension
+  end
+
+  def output_path()
+    raise NotGeneratedError unless @generated_at
```
Comment (jesusbv) [inline app/models/supportconfig.rb:39]: ditto

File: bin/supportconfig [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  when docopt['-g']
+    'tgz'
+  else
+    'txz'
+  end
+  filename = 'scc_' + docopt['-B'] + '.' + extension
+  full_path = Pathname.new(docopt['-R']).join(filename)
+
+  FileUtils.touch(full_path)
+rescue Docopt::Exit => e
```
Comment (jesusbv) [inline bin/supportconfig:38]: should we handle if path is wrong ?  adding ` ArgumentError` (`Pathname` error raised) to the rescue block ?

File: app/handlers/exceptions_app.rb (hunk 3 above)
Comment (bear454) [reply app/handlers/exceptions_app.rb:12]: All exceptions inherit from StandardError

File: app/models/concerns/authenticateable.rb (hunk 2 above)
Comment (bear454) [reply app/models/concerns/authenticateable.rb:23]: stdout & stderr is only populated if there is no exception (e.g. exit code 0). It's unlikely there's anything useful in stderr.

File: bin/supportconfig (hunk 7 above)
Comment (bear454) [reply bin/supportconfig:38]: This isn't `supportconfig` - it's just a mock to handle execution in development. It meets the CLI expectation; I don't plan to add a robust CLI to a mock.

File: app/models/concerns/authenticateable.rb (hunk 1 above)
Comment (bear454) [reply app/models/concerns/authenticateable.rb:16]: It's not hurting anything

File: app/models/supportconfig.rb (hunk 6 above)
Comment (bear454) [reply app/models/supportconfig.rb:40]: 6759926

File: app/models/supportconfig.rb (hunk 5 above)
Comment (bear454) [reply app/models/supportconfig.rb:34]: 6759926

**Comment by bear454** [comment]
I don't expect a deep review here; everything was reviewed in a PR to the next branch except the changelog. We just need a quick once-over to make sure it's good to go.

File: app/models/concerns/authenticateable.rb (hunk 2 above)
Comment (jesusbv) [reply app/models/concerns/authenticateable.rb:23]: OK, so `openssl passwd` does not populate `stderr` cool

File: app/models/supportconfig.rb (hunk 4 above)
Comment (jesusbv) [reply app/models/supportconfig.rb:14]: is it the same? supportconfig does not populate `stderr` ?

**Review by jesusbv — APPROVED** [review]

**Pull request merged**